gateway/gateway.go

```diff
@@ -922,22 +922,25 @@ func (gateway *HandleT) pendingEventsHandler(w http.ResponseWriter, r *http.Requ
 		}
 	}
 
+	ctx, cancel := context.WithTimeout(context.TODO(), config.GetDuration("Gateway.pendingEventsQueryTimeout", time.Duration(10), time.Second))
```

Comment:
We could get the context for the [request](https://pkg.go.dev/net/http#Request.Context):

```suggestion
	ctx, cancel := context.WithTimeout(r.Context(), config.GetDuration("Gateway.pendingEventsQueryTimeout", time.Duration(10), time.Second))
```

The benefit is HTTP request timeouts or request getting canceled will also affect the context.